Update Antivirus forced by App update for ICAP (#746)
owncloud/docs-server: modules/admin_manual/pages/configuration/server/virus-scanner-support.adoc

    av_scan_background --value="true"
----

. Specify what to do with the offending file:
+
[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set files_antivirus \
    av_infected_action --value="delete"
----
+
Possible values are `delete` and `only_log`.
+
Note that this setting is only used when `av_scan_background` is set to `true`.

==== Mode Dependent Settings

Following settings are just an example and must be set according the scanner used. For possible values see xref:configure-the-antivirus-app-using-occ[Configure the Antivirus App Using occ]

. Set the ICAP mode:
+
[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set files_antivirus \
    av_mode --value="icap"
----

. Set the respective request service:
+
[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set files_antivirus \
    av_request_service --value="avscan"
----

. Set the respective response header:
+
[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set files_antivirus \
    av_response_header --value="X-Infection-Found"
----

=== ClamAV & Kaspersky (ICAP)

> Specific for ClamAV, you first must install and configure {c-icap-url}[c-icap on sourceforge]. For selecting ClamAV in their configuration see the section: https://sourceforge.net/p/c-icap/wiki/ModulesConfiguration/[ModulesConfiguration].

Reviewer: ```suggestion
If you decide to use ClamAV, you must first install and configure {c-icap-url}[c-icap on sourceforge]. With regard to selecting ClamAV in their configuration, see the section: https://sourceforge.net/p/c-icap/wiki/ModulesConfiguration/[ModulesConfiguration].
```

. Install ClamAV based on the instructions at the beginning of this document and `c-icap` as referenced above.

> . Select btn:[ClamAV & Kaspersky (ICAP)] from the dropdown in the Anti Virus app:

Reviewer: ```suggestion
. Select btn:[ClamAV & Kaspersky (ICAP)] from the drop-down menu in the Anti-Virus app:
```

+
image::configuration/server/virus-scanner-support/icap_clamav.png[ClamAV & Kasperski (ICAP)],width=500]

NOTE: The older versions of Kaspersky’s KAV did not send back the virus/infection name in an ICAP header. Starting with KAV v2.0.0, the header to transport the virus can be configured. Default: No header is sent. For more configuration details see {kaspersky-icap-url}[Using Kaspersky Scan Engine in ICAP mode].

// note that the original document regarding "VirusNameICAPHeader" https://support.kaspersky.com/ScanEngine/1.0/en-US/201214.htm is not available anymore (404, no cache entry) and no appropriate replacement has been found. the link above is the best possible alternative.

=== McAfee Webgateway / Skyhigh Secure Webgateway (ICAP)

> Follow this procedure to configure ownCloud for the McAfee Web Gateway 10.x and higher in ICAP Mode.

Reviewer: ```suggestion
Follow the procedure below to configure ownCloud for the McAfee Web Gateway 10.x and higher in ICAP Mode.
```

. Install McAfee Web Gateway based on their instructions.

> . Select btn:[McAfee Webgateway / Skyhigh Secure Webgateway (ICAP)] from the dropdown in the Anti Virus app.

Reviewer: ```suggestion
. Select btn:[McAfee Web Gateway / Skyhigh Secure Web Gateway (ICAP)] from the drop-down menu in the Anti-Virus app.
```

+

> image::configuration/server/virus-scanner-support/icap_mcafee.png[McAfee Webgateway / Skyhigh Secure Webgateway (ICAP)],width=500]

Reviewer: ```suggestion
image::configuration/server/virus-scanner-support/icap_mcafee.png[McAfee Web Gateway / Skyhigh Secure Web Gateway (ICAP)],width=500]
```

NOTE: McAfee provides demo versions with limited runtime for evaluation purposes. Have a look at the {mcaffee-demo-url}[McAfee Web page] for the Web Gateway.

=== Fortinet (ICAP)

. Install FortiSandbox based on their instructions.

> . Select btn:[Fortinet] from the dropdown in the Anti-Virus app.

Reviewer: ```suggestion
. Select btn:[Fortinet] from the drop-down menu in the Anti-Virus app.
```

+

> image::configuration/server/virus-scanner-support/icap_mcafee.png[McAfee Webgateway / Skyhigh Secure Webgateway (ICAP)],width=500]

Reviewer: ```suggestion
image::configuration/server/virus-scanner-support/icap_mcafee.png[McAfee Web Gateway / Skyhigh Secure Web Gateway (ICAP)],width=500]
```
